journal.py

```diff
@@ -74,6 +138,16 @@ def db_exception(context, request):
     return response
 
 
+@view_config(route_name='permalink', renderer="templates/permalink.jinja2")
+def permalink_view(request):
+    entry = Entry.id_lookup(request.matchdict['id'])
+    return {'entry': {
+            'id': entry.id,
+            'title': entry.title,
+            'text': entry.text,
+            'created': entry.created}}
```

Comment:
if you just pass the entry object itself in to the template, rather than this dictionary, you wouldn't have to change your template at all.  All the keys in this dict are also attribute names of the entry object.